tutorials/appengine-ruby-rails-activejob-pubsub.md

```diff
@@ -0,0 +1,272 @@
+
+1. Add a route for Google App Engine health checks. In `config/routes.rb`, add:
+
+        get '/_ah/health', to: proc { [200, {}, ['ok']]}
+
+   Without this route, Google App Engine will think your application is crashing and will not send any traffic to it.
+
+## Configuring your background worker to use Cloud Pub/Sub
+
+ActiveJob can be configured with various different background job runners. This tutorial will cover
+[ActiveJob::GoogleCloudPubsub](https://github.com/ursm/activejob-google_cloud_pubsub) which uses Google Cloud Pub/Sub
+to manage the job queue.
+
+1. Add `activejob-google_cloud_pubsub` gem to your `Gemfile`:
+
+        bundle add activejob-google_cloud_pubsub
+
+1. Configure ActiveJob to use GoogleCloudPubsub as its queue adapter. In `config/application.rb`:
+
+        class Application < Rails::Application
+          # ...
+          config.active_job.queue_adapter = :google_cloud_pubsub
+        end
+
+## Deploying to App Engine Flexible Environment
+
+### Option A: Shared worker and web application
+
+For this option, the App Engine service will run both the web server and a worker process via a process manager called
+[foreman](https://ddollar.github.io/foreman/). If you choose this method, App Engine will scale your web and worker
+instances together.
+
+1. Add `foreman` gem to your `Gemfile`:
+
+        bundle add foreman
+
+1. Create a `Procfile` at the root of your application:
+
+        web: bundle exec rails server -p 8080
+        worker: bundle exec activejob-google_cloud_pubsub-worker
+
+1. Create an `app.yaml` for deploying the application to Google App Engine:
+
+        runtime: ruby
+        env: flex
+
+        entrypoint: bundle exec foreman start
+
+        env_variables:
+          SECRET_KEY_BASE: [SECRET_KEY]
+
+   Be sure to replace the `[SECRET_KEY]` with a secret key for Rails sessions.
+
+1. Deploy to App Engine
+
+        gcloud app deploy app.yaml
+
+### Option B: Separate worker and web application
+
+For this option, you are creating 2 App Engine services - one runs the web server and one runs worker processes. Both
+services use the same application code. This configuration allows you to scale background workers independently of
+your web workers at the cost of potentially using more resources.
+
+1. Create an `app.yaml` for deploying the web service to Google App Engine:
+
+        runtime: ruby
+        env: flex
+
+        entrypoint: bundle exec rails server -p 8080
+
+        env_variables:
+          SECRET_KEY_BASE: [SECRET_KEY]
+
+   Be sure to replace the `[SECRET_KEY]` with a secret key for Rails sessions.
+
+1. Create a `workers.yaml` for deploying the workers service to Google App Engine:
+
+        runtime: ruby
+        env: flex
+        service: workers
+
+        entrypoint: activejob-google_cloud_pubsub-worker
+
+        env_variables:
+          SECRET_KEY_BASE: [SECRET_KEY]
+
+        health_check:
+          enable_health_check: False
```

Comment:
I think it would be better to specify `manual_scaling` here. `automatic_scaling` implies we're using the [autoscaler](https://cloud.google.com/appengine/docs/flexible/ruby/an-overview-of-app-engine#instance_scaling) which measures incoming requests, etc., and since there are none for a worker, that really seems confusing. (I actually have no idea how the autoscaler would behave for such a workload.)

Same for the other tutorial.